### PR TITLE
SW-4484 Update Filters from URL Search Params

### PR DESCRIPTION
**Pull request by charlie-tf**
- Add SubLocations Redux
- In the InventoryListByBatch page get filters from query
- Look up sublocations and apply filters
- When filters change update the URL query
Kapture 2023-12-01 at 16 34 38

### CODE REVIEW

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 1]
```diff
@@ -86,6 +92,56 @@ export default function InventoryListByBatch({ setReportData }: InventoryListByB
...
+  }, [nurseryIds, setFilters]);
+
+  useEffect(() => {
+    dispatch(requestSubLocations(filters.facilityIds ?? []));
+  }, [filters.facilityIds, dispatch]);
+
+  useEffect(() => {
+    const subLocationsIds = subLocationNames
+      .map((name) => subLocations?.find((sl) => sl.name === name)?.id)
+      .filter((id) => id !== undefined) as number[];
```
Comment (nickgraz) [inline src/components/InventoryV2/InventoryListByBatch.tsx:127]: You might be able to use a typeguard on the filter which would remove the need to cast.
```suggestion
      .filter((id): id is number => id !== undefined);
```

File: src/components/InventoryV2/Search.tsx [hunk 2]
```diff
@@ -63,25 +63,12 @@ export default function Search(props: SearchProps): JSX.Element | null {
...
-      setSubLocations(results.flatMap((r) => r.subLocations));
-    };
-    if (!filters.facilityIds?.length) {
-      setSubLocations([]);
-    } else {
-      fetchSubLocations();
-    }
-    setFilters({ ...filters, subLocationsIds: [] });
-  }, [filters, setFilters]);
+  const subLocations = useAppSelector(selectSubLocations);
```
Comment (nickgraz) [inline src/components/InventoryV2/Search.tsx:71]: 🔥

**Review by nickgraz — APPROVED** [review]
Looks good, one small thing to try to get rid of a cast but not a blocker

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 3]
```diff
@@ -86,6 +92,56 @@ export default function InventoryListByBatch({ setReportData }: InventoryListByB
...
+      ?.filter((s) => s !== '');
+    setSubLocationNames(querySubLocationNames ?? []);
+    const queryNurseryIds = query
+      .get('facilityId')
+      ?.trim()
+      ?.split(',')
+      ?.filter((n) => n !== '')
+      ?.map((id) => Number(id));
+    setNurseryIds(queryNurseryIds ?? []);
+  }, [query]);
```
Comment (karthikbtf) [inline src/components/InventoryV2/InventoryListByBatch.tsx:114]: Thanks for adding this feature @charlie-tf ❤️ 
Added a comment in the Jira, supporting multiple params is better done this [way](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#duplicate_search_parameters) and using `query.getAll(paramName)`

**Pull request merged**